app/src/components/common/Table.vue

```diff
@@ -44,21 +61,90 @@
 </template>
 
 <script lang="ts">
-import { TableColumn } from '@/types'
+import { TableColumn, SortOrder, SortRule } from '@/types'
 import { Vue, Component, Prop } from 'vue-property-decorator'
 
-@Component
+@Component({
+  computed: {
+    SortOrder () {
+      return SortOrder
+    }
+  }
+})
 export default class Table extends Vue {
   @Prop({ required: true }) columns!: TableColumn[]
   @Prop({ required: true }) data!: Array<{ [key: string]: string }>
+  sortedData: Array<{ [key: string]: string }> = []
+  sortingColumn: TableColumn | undefined = undefined
+  sortingRules: SortRule[] = [
+    {
+      currSort: SortOrder.NEUTRAL,
+      nextSort: SortOrder.ASC
+    },
+    {
+      currSort: SortOrder.ASC,
+      nextSort: SortOrder.DESC,
+      compareFn: (firstObjectToSort: any, secondObjectToSort: any) =>
+          typeof firstObjectToSort === 'string' && typeof secondObjectToSort === 'string'
+              ? firstObjectToSort.toLowerCase().localeCompare(secondObjectToSort.toLowerCase())
+              : firstObjectToSort > secondObjectToSort ? 1 : -1
+    },
+    {
+      currSort: SortOrder.DESC,
+      nextSort: SortOrder.NEUTRAL,
+      compareFn: (firstObjectToSort: any, secondObjectToSort: any) =>
+          typeof firstObjectToSort === 'string' && typeof secondObjectToSort === 'string'
+              ? secondObjectToSort.toLowerCase().localeCompare(firstObjectToSort.toLowerCase())
+              : firstObjectToSort > secondObjectToSort ? -1 : 1
+    }
+  ]
+
+  mounted () {
+    this.columns.forEach((column) => {
+      if (column.isSortable && !column.sortOrder) {
+        this.$set(column, 'sortOrder', SortOrder.NEUTRAL)
+      }
+    })
+  }
 
   getLabel (column: TableColumn): string {
-    return column.label ? column.label : ''
+    return column.label || ''
   }
 
   getEntry (column: TableColumn, item: { [key: string]: string }): string {
     return item[ column.key ]
   }
+
+  setSortingColumn (column: TableColumn) {
+    if (column.isSortable) {
+      this.sortingColumn = column
+      column.sortOrder = this.sortingRules.find(
+          (rule) => rule.currSort === column.sortOrder
+      )!.nextSort
+      this.updateSortedData()
+    }
+  }
+
+  get displayData (): Array<any> {
+    this.sortedData = this.data.slice()
+    this.updateSortedData()
```

Comment:
not quite clear why `updateSortedData()` is called here?

Reply:
the updateSortedData is called here so that if data changes (if it's loaded or edited etc) we can update display data accordingly and sort data by current sorting column and sorting direction